diff --git a/hwloc/topology-linux.c b/hwloc/topology-linux.c
index 0842b0b..dec6bcc 100644
--- a/hwloc/topology-linux.c
+++ b/hwloc/topology-linux.c
@@ -106,6 +106,8 @@ struct hwloc_linux_backend_data_s {
 #         define __NR_sched_setaffinity 241
 /*#       elif defined(__mips__)
   #         define __NR_sched_setaffinity TODO (32/64/nabi) */
+#       elif defined(__sw_64__)
+#         define __NR_sched_setaffinity 395
 #       else
 #         warning "don't know the syscall number for sched_setaffinity on this architecture, will not support binding"
 #         define sched_setaffinity(pid, lg, mask) (errno = ENOSYS, -1)
@@ -143,6 +145,8 @@ struct hwloc_linux_backend_data_s {
 #         define __NR_sched_getaffinity 242
 /*#       elif defined(__mips__)
   #         define __NR_sched_getaffinity TODO (32/64/nabi) */
+#       elif defined(__sw_64__)
+#         define __NR_sched_getaffinity 396
 #       else
 #         warning "don't know the syscall number for sched_getaffinity on this architecture, will not support getting binding"
 #         define sched_getaffinity(pid, lg, mask) (errno = ENOSYS, -1)
